lang/cs/Org.Apache.REEF.Wake.Tests/DefaultTcpPortProviderFactory.cs

```diff
@@ -0,0 +1,80 @@
+﻿/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Org.Apache.REEF.Wake.Remote;
+using Org.Apache.REEF.Wake.Remote.Parameters;
+
+namespace Org.Apache.REEF.Wake.Tests
+{
+    /// <summary>
+    /// Creates new instances of ITcpPortProvider.
+    /// </summary>
+    public class DefaultTcpPortProviderFactory : ITcpPortProviderFactory
+    {
+        [Inject]
+        public DefaultTcpPortProviderFactory()
```

Comment:
Actually, why do we need a factory for this at all? `TcpPortProvider` is a singleton for the process, right?